From 74feb0759484202496e7639c5645e49a7258c9d5 Mon Sep 17 00:00:00 2001
From: Antonio Ojea <antonio.ojea.garcia@gmail.com>
Date: Wed, 12 May 2021 19:15:56 +0200
Subject: [PATCH] kubeadm: CoreDNS permissions for endpointslices

This patch is resolves an issue related to EKS-D being ahead of
upstream Kubernetes in version of CoreDNS use, which EKS-D issue #545
(https://github.com/aws/eks-distro/issues/545) raised. This change is
a partial cherry-pick of upstream Kubernetes PR #102466
(https://github.com/kubernetes/kubernetes/pull/102466

Signed-off-by: Antonio Ojea <antonio.ojea.garcia@gmail.com>
Co-authored-by: pacoxu <paco.xu@daocloud.io>
---
 cmd/kubeadm/app/phases/addons/dns/manifests.go | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/cmd/kubeadm/app/phases/addons/dns/manifests.go b/cmd/kubeadm/app/phases/addons/dns/manifests.go
index 3ac6856bfc6..97c7f8b3e60 100644
--- a/cmd/kubeadm/app/phases/addons/dns/manifests.go
+++ b/cmd/kubeadm/app/phases/addons/dns/manifests.go
@@ -197,6 +197,13 @@ rules:
   - nodes
   verbs:
   - get
+- apiGroups:
+  - discovery.k8s.io
+  resources:
+  - endpointslices
+  verbs:
+  - list
+  - watch
 `
 	// CoreDNSClusterRoleBinding is the CoreDNS Clusterrolebinding manifest
 	CoreDNSClusterRoleBinding = `
-- 
2.25.1
